selfdrive/controls/lib/lateral_planner.py

```diff
@@ -173,12 +173,12 @@ def update(self, sm, CP):
       # Heading cost is useful at low speed, otherwise end of plan can be off-heading
       heading_cost = interp(v_ego, [5.0, 10.0], [MPC_COST_LAT.HEADING, 0.0])
       self.libmpc.set_weights(path_cost, heading_cost, CP.steerRateCost)
-    y_pts = np.interp(v_ego * self.t_idxs[:MPC_N + 1], np.linalg.norm(d_path_xyz, axis=1), d_path_xyz[:,1])
-    heading_pts = np.interp(v_ego * self.t_idxs[:MPC_N + 1], np.linalg.norm(self.path_xyz, axis=1), self.plan_yaw)
+    y_pts = np.interp(v_ego * self.t_idxs[:LAT_MPC_N + 1], np.linalg.norm(d_path_xyz, axis=1), d_path_xyz[:,1])
```

Comment:
Try using `np_fast.interp` on arrays that aren't numpy arrays already. That can have a measurable performance improvement.

Reply:
Yeah will do performance at the end. Still quite a few things that are not done. I will write to do list

Reply:
Sounds good to e